src/commons/product_image_selection.rs

```diff
@@ -0,0 +1,347 @@
+use k8s_openapi::api::core::v1::LocalObjectReference;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use strum::AsRefStr;
+
+pub const STACKABLE_DOCKER_REPO: &str = "docker.stackable.tech/stackable";
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ProductImage {
+    #[serde(flatten)]
+    image_selection: ProductImageSelection,
+
+    #[serde(default)]
+    /// [Pull policy](https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy) used when pulling the Images
+    pull_policy: PullPolicy,
+
+    /// [Image pull secrets](https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod) to pull images from a private registry
+    pull_secrets: Option<Vec<LocalObjectReference>>,
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(untagged)]
+pub enum ProductImageSelection {
+    // Order matters!
+    // The variants will be tried from top to bottom
+    Custom(ProductImageCustom),
+    StackableVersion(ProductImageStackableVersion),
+    // The following enum variant is commented out for now, as the operators currently don't know which stackableVersions they are compatible with.
+    // In the future they should be able to automatically pick a fitting stackableVersion for them.
+    // The concept of the untagged enum was tested with all known upcoming variants to make sure we don't run into strange problems in the future.
+    // They code snippets are left for illustration and can be commented in as soon as the operators can pick the stackableVersions automatically.
+    Stackable(ProductImageStackable),
+}
+
+impl Default for ProductImageSelection {
```

Comment:
Doesn't look like this `Default` impl is used anywhere?

Reply:
Yep, that's correct. Was a leftover, removed it